cmd/report.go

```diff
@@ -84,16 +110,181 @@ var (
+			strBuilder.WriteString("\nERROR: could not download cookbook (see end of report)")
+			downloadErrors.WriteString(fmt.Sprintf(" - %s (%s): %v\n", record.Name, record.Version, record.DownloadError))
+		} else if record.CookstyleError != nil {
+			strBuilder.WriteString("\nERROR: could not run cookstyle (see end of report)")
+			cookstyleErrors.WriteString(fmt.Sprintf(" - %s (%s): %v\n", record.Name, record.Version, record.CookstyleError))
+		} else if record.UsageLookupError != nil {
+			strBuilder.WriteString("\nERROR: unknown violations (see end of report)")
+			usageFetchErrors.WriteString(fmt.Sprintf(" - %s (%s): %v\n", record.Name, record.Version, record.UsageLookupError))
+		}
+
+		// TODO @afiune write report to disk
+		fmt.Println(strBuilder.String())
+	}
+
+	writeErrorBuilders(downloadErrors, cookstyleErrors, usageFetchErrors)
+}
+
+func writeDetailedCookbookStateReport(records []*reporting.CookbookRecord) {
+	var (
+		downloadErrors   strings.Builder
+		usageFetchErrors strings.Builder
+		cookstyleErrors  strings.Builder
+	)
+	for _, record := range records {
+		var strBuilder strings.Builder
+
+		// skip unused cookbooks
+		if len(record.Nodes) == 0 && cookbooksFlags.skipUnused {
+			continue
+		}
+
+		strBuilder.WriteString(fmt.Sprintf("%v (%v) ", record.Name, record.Version))
+		strBuilder.WriteString(fmt.Sprintf("%v violations, %v auto-correctable, %v nodes affected",
+			record.NumOffenses(), record.NumCorrectable(), len(record.Nodes)),
+		)
+
+		strBuilder.WriteString("\nNodes affected: ")
+		if len(record.Nodes) == 0 {
+			strBuilder.WriteString("none")
+		} else {
+			strBuilder.WriteString(strings.Join(record.Nodes, ", "))
+		}
+		strBuilder.WriteString("\nFiles and offenses:")
+		for _, f := range record.Files {
+			if len(f.Offenses) == 0 {
+				continue
+			}
+			strBuilder.WriteString(fmt.Sprintf("\n - %s:", f.Path))
+			for _, o := range f.Offenses {
+				strBuilder.WriteString(fmt.Sprintf("\n\t%s (%t) %s", o.CopName, o.Correctable, o.Message))
+			}
+		}
+
+		if record.DownloadError != nil {
+			strBuilder.WriteString("\nERROR: could not download cookbook (see end of report)")
+			downloadErrors.WriteString(fmt.Sprintf(" - %s (%s): %v\n", record.Name, record.Version, record.DownloadError))
+		} else if record.CookstyleError != nil {
+			strBuilder.WriteString("\nERROR: could not run cookstyle (see end of report)")
+			cookstyleErrors.WriteString(fmt.Sprintf(" - %s (%s): %v\n", record.Name, record.Version, record.CookstyleError))
+		} else if record.UsageLookupError != nil {
+			strBuilder.WriteString("\nERROR: unknown violations (see end of report)")
+			usageFetchErrors.WriteString(fmt.Sprintf(" - %s (%s): %v\n", record.Name, record.Version, record.UsageLookupError))
+		}
+
+		// TODO @afiune write report to disk
+		fmt.Println(strBuilder.String())
+	}
+
+	writeErrorBuilders(downloadErrors, cookstyleErrors, usageFetchErrors)
+}
+
+func writeDetailedCSV(records []*reporting.CookbookRecord) {
```

Comment:
I originally tried to be fancy here and I think it is a mistake. Instead of outputting empty cells I should instead have every cell in every row populated. We can come back for that in a follow up PR.